Sources/JavaRuntime/AndroidSupport.cpp

```diff
@@ -34,7 +34,7 @@ static JavaRuntime_GetCreatedJavaVMs_fn JavaRuntime_GetCreatedJavaVMs;
 static void *JavaRuntime_dlhandle;
 
 __attribute__((constructor)) static void JavaRuntime_init(void) {
-  JavaRuntime_dlhandle = dlopen("libnativehelper.so", RTLD_NOW | RTLD_GLOBAL);
+  JavaRuntime_dlhandle = dlopen("libnativehelper.so", RTLD_NOW | RTLD_LOCAL);
```

Comment:
Question about this! Should we link the JavaRuntime target to `libnativehelper.so` when building for Android rather than doing the `dlopen` here?

Reply:
Hmm, I'd assumed this wasn't possible because the Android NDK only provides the `.so`, and typically on Linux the linker will only find a versioned `.so`. But maybe this isn't the case on Android. I'll check and get back.

Reply:
Actually I may have that around the wrong way, you need the unversioned `.so` at `ld` time, and the versioned one at runtime. So yes, checking I can link against it now.

Reply:
OK, this is possible, but requires setting the NDK target to >=31:

```
        .unsafeFlags(["--target=aarch64-unknown-linux-android31"], .when(platforms: [.android])),
```

Obviously I don't want to encode the target architecture, do you know offhand if this is surfaced?

Reply:
* Actually this is a bit harder than it seems, for reasons I'll explain in a comment later.